target/linux/lantiq/dts/RS353.dts

```diff
@@ -0,0 +1,318 @@
+/dts-v1/;
+
+/include/ "vr9.dtsi"
+
+/ {
+	model = "RS353 - Bintec Elmeg RS353";
+
+	chosen {
+		bootargs = "console=ttyLTQ0,115200 init=/etc/preinit";
+		leds {
+                        running = &status;
+                        boot = &status;
+			wifi = &wlan;
+			usb2 = &usb; /* Front port is mapped to device 2 */
+			dsl = &dsl;
+                };
+
+	};
+
+	memory@0 {
+		reg = <0x0 0x8000000>;
+	};
+
+	sram@1F000000 {
+		cgu0: cgu@103000 {
+                        compatible = "lantiq,cgu-xway";
+                        reg = <0x103000 0x1000>;
+                        #clock-cells = <1>;
+			lantiq,phy-clk-src = <0x2>;
+		};
+	};
+
+	fpi@10000000 {
+		localbus@0 {
+			nor-boot@0 {
+				compatible = "lantiq,nor";
+				bank-width = <2>;
+				reg = <0 0x0 0x2000000>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+
+				partitions {
+					compatible = "fixed-partitions";
+					#address-cells = <1>;
+					#size-cells = <1>;
+
+					boardconfig: partition@0 {
```

Comment:
My idea is to replace the Bootmonitor by a 128k fake image containing just a jump to the kernel loader.

The advantage is the image can be flashed by the WEB interface, and the first bootloader calls directly the Linux kernel. But it is harder to come back to the original firmware. 

Another idea is to flash the sysupgrade image as BOSS, but as you said the CRC will be incorrect after the first boot. So, I have to find a way to adjust the CRC after first boot.

The gzipped part will contain only the 128k image containing the ASM jump to the image. As you said, I kept the "u-boot" partitions, but I will drop it. My hex code to jump to the right offset is the following : 

```
3C 08 10 12
21 08 00 40
01 00 00 08
```
I load 0x10120040 flash addr into t0 register, and I jump to it. I can call it "preloader" or "jumper".

Regarding the flash endianness issue, I guess the bootmonitor has some kind of workaround to fix the lantiq bug ? The pcie port works well, as you can see in my boot log. I even tried a sysupgrade procedure, which works well. The advantage of replacing the Bootmonitor by my jump code is the system can be upgraded via sysupgrade without any difficult. 

Regarding the uart issue, I would like to make a separated pull request  as I guess this is a real bug (the lack of waiting before sending the '\n' make it dropped).